pkg/asset/installconfig/azure/metadata.go

```diff
@@ -19,12 +19,24 @@ type Metadata struct {
 	// CloudName indicates the Azure cloud environment (e.g. public, gov't).
 	CloudName typesazure.CloudEnvironment `json:"cloudName,omitempty"`
 
+	// Credentials hold prepopulated Azure credentials.
+	// At the moment the installer doesn't use it and reads credentials
+	// from the file system, but external consumers of the package can
+	// provide credentials. This is useful when we run the installer
+	// as a service (Azure Red Hat OpenShift, for example): in this case
+	// we do not want to rely on the filesystem or user input as we
+	// serve multiple users with different credentials via a web server.
+	Credentials *Credentials `json:"credentials,omitempty"`
```

Comment:
For the installer this field will be omited during marshaling as the installer doesn't use it. 

Wondering if we can avoid marshaling, but my previous implementation (like [this one with an interface](https://github.com/openshift/installer/commit/de541bf88a6bda8a294a944f788f2e5d7266277f)) failed.

Have I missed any implications of doing this?